File: backend/src/main/java/com/backend/api/user/init/UserDataSyncRunner.java
package com.backend.api.user.init;

import com.backend.domain.user.entity.User;
import com.backend.domain.user.entity.search.UserDocument;
import com.backend.domain.user.repository.UserRepository;
import com.backend.domain.user.repository.search.UserSearchRepository;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;

import java.util.List;

@Slf4j
@Component
@RequiredArgsConstructor
public class UserDataSyncRunner implements CommandLineRunner {

    private final UserRepository userRepository;
    private final UserSearchRepository userSearchRepository;

    @Override
    public void run(String... args) {
        List<User> users = userRepository.findAll();

        List<UserDocument> docs = users.stream()
                .map(u -> UserDocument.builder()
                        .id(u.getId().toString())
                        .name(u.getName())
                        .nickname(u.getNickname())
                        .email(u.getEmail())
                        .build())
                .toList();

        userSearchRepository.saveAll(docs);
        log.info("Elasticsearch 인덱싱 완료: {}건 저장됨", docs.size());
    }
}

File: backend/src/main/java/com/backend/api/user/init/UserDataSyncRunner.kt
package com.backend.api.user.init

import com.backend.domain.user.entity.search.UserDocument
import com.backend.domain.user.repository.UserRepository
import com.backend.domain.user.repository.search.UserSearchRepository
import org.slf4j.LoggerFactory
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component

@Component
class UserDataSyncRunner(
    private val userRepository: UserRepository,
    private val userSearchRepository: UserSearchRepository
) : CommandLineRunner {

    private val log = LoggerFactory.getLogger(UserDataSyncRunner::class.java)

    override fun run(vararg args: String?) {
        val users = userRepository.findAll()

        val docs = users.map { u ->
            UserDocument.from(u)
        }

        userSearchRepository.saveAll(docs)
        log.info("Elasticsearch 인덱싱 완료: {}건 저장됨", docs.size)
    }
}
